proxy.go

```diff
@@ -47,15 +47,39 @@ var sandboxID string
 
 var proxyLog = logrus.New()
 
+// This function is meant to run in a go routine since it will send ping
+// commands every second. It behaves as a heartbeat to maintain a proper
+// communication state with the Yamux server in the agent.
+func heartBeat(session *yamux.Session) {
```

Comment:
yamux `config.KeepAliveInterval` already do this,  we just need to increase `config.ConnectionWriteTimeout`